### ISSUE TITLE
Window Says No Internet

### ISSUE THREAD

**Issue by sutex** [bitbeans/SimpleDnsCrypt#493]
If I set the DNS to 127.0.0.1 and the IP to automatic.
I get the world globe saying not connected, yet can access the net.

If I change to a Static IP on the adapter and the DNS is 127.0.0.1
I get the world globe saying not connected, yet can access the net.

Sometime I have no internet access at all, when setting the DNS to 127.0.0.1

If I run the Troubleshooter, it resets the DNS and the internet icon changes to connected, and internet runs normally.

[2020-04-19 13_03_20-Window](https://user-images.githubusercontent.com/10361359/79678919-87232780-823f-11ea-89ed-abd79cdf5e80.png)

**Comment by sutex** [comment]
I Updated the drive "PROWinx64.exe" for the adapter Intel(R) Ethernet Connection (2) I219-V for Win 10 64bit. 

Windows says it's connected to the internet and the "Globe icon" to changes to connected when using automatic IP and DNS 127.0.01 This happens immediately.

When using a Static IP on the adapter and the DNS is 127.0.0.1 with the new driver, 15minutes may pass before Windows confirms it's connected to internet and the Globe icon to changes to connected

**Comment by sutex** [comment]
Update the issue is still the same, given further time ruining the new PROWinx64.exe driver.

As soon as the the DNS is set to 127.0.0.1, the Icon changes to a Globe and windows thinks it's not connected to the internet. It is completly random if the icon changes and window reports via setting that the PC is a connected to the internet. 

Using my default ISP DNS windows connects and shows the connection normally

**Comment by ledlamp** [comment]
I started having this issue after upgrading to version 2004 (build 19041.208). I had to uninstall the windows service.

**Comment by jedisct1** [comment]
Maybe this can help: https://github.com/dnscrypt/dnscrypt-proxy/wiki/Installation-Windows#troubleshooting

**Comment by sutex** [comment]
Yes, the reg key seems to have done the trick, On reboot Icon changed to connected within 15 seconds.

**Comment by LoLyeah** [comment]
I have this same problem too, and this problem happened after I updated my Windows 10 to version 2004. I'll try the reg key trick above first.

**Comment by sutex** [comment]
The Reg 'trick' above worked on my Win 10 build 1904 18363.815. Yet to upgrade

**Comment by edwardqiu** [comment]
With no blacklist set, I had the same issue where Windows' network status was displaying `No Internet access`, but all applications were making DNS queries and getting responses, except Spotify's desktop client.

I had Spotify's desktop client with the `No Proxy` setting and it would display it was offline with error code 4 on startup, which would block playing any music that wasn't downloaded already.

Adding the registry key fixed both issues for me on Windows 10 1909 18363.815

Just putting a comment here in case anyone else is running into the same issue with Spotify on Windows

**Comment by jmacasd** [comment]
I also have this same issue after upgrading to 2004.

**Comment by MoneyAllDay** [comment]
I wonder if all the people in this topic [Windows 10 says I have no internet connection, but I have.](https://answers.microsoft.com/en-us/windows/forum/all/windows-10-says-i-have-no-internet-connection-but/50153e5b-a008-443d-af0e-5ee3822c37a1) are also using DnsCrypt.

I also never had this issue before Win10 2004 update.
